### DISCUSSION TITLE
Home Assistant MyQ garage workaround

### DISCUSSION THREAD

**Discussion by mgrimace** [tteck/Proxmox#1781] [General]
Hi, sharing this here in case others are using any of the Home Assistant scripts and experiencing the MyQ garage outage ([issue](https://github.com/home-assistant/core/issues/99947)): 

I'm using the HAOS vm, and here are the tentative workaround steps:

1. Install the advanced SSH and web terminal addon, found here - https://github.com/hassio-addons/addon-ssh/blob/main/ssh/DOCS.md
2. Set a username and password in the config of the addon, and disable protection mode. 
3. Start or restart the addon
4. SSH using the webui for the addon in Home Assistant, or using your ssh client
5. enter the following commands into your SSH terminal:

```bash
docker exec -it homeassistant bash
cd /usr/local/lib/python3.11/site-packages/pymyq/
cp request.py request_bup.py
vi request.py
i to enter insert mode
Modify line 34

    Was self._useragent = None
    Is self._useragent = str("myq-ios-5.240.0.38162")
```

6. restart home assistant

**Comment by mgrimace** [comment]
Alternatively, Homebridge with the MyQ addon seems to be working fine

**Comment by tteck** [comment]
I also saw this:
```
docker exec -it homeassistant sed -i.bak "34s/None/str(\"ha${RANDOM}\")/" /usr/local/lib/python3.11/site-packages/pymyq/request.py
```
Essentially, it accomplishes what you've recommended, but it streamlines the process by utilizing `sed`.

**Comment by btehan** [comment]
Only tangentially related, but I've used this to move away from MyQ:
https://paulwieland.github.io/ratgdo/

**Reply by mgrimace** [reply]
That's intriguing, thanks so much for putting this on my radar!
